processing/src/main/java/org/apache/druid/segment/transform/TransformSpec.java

```diff
@@ -44,19 +44,23 @@
  */
 public class TransformSpec
 {
-  public static final TransformSpec NONE = new TransformSpec(null, null);
+  public static final TransformSpec NONE = new TransformSpec(null, null, null);
 
   private final DimFilter filter;
   private final List<Transform> transforms;
+  private final List<String> addDimensions;
+
 
   @JsonCreator
   public TransformSpec(
       @JsonProperty("filter") final DimFilter filter,
-      @JsonProperty("transforms") final List<Transform> transforms
+      @JsonProperty("transforms") final List<Transform> transforms,
+      @JsonProperty("addDimensions") final List<String> addDimensions
```

Comment:
This seems like it could work, without the downside of always including them as dimensions. The alternative I guess would be adding a `isDimension` to the `Transform` interface? I'm not sure which is better, though the way you've done it here is definitely less intrusive in terms of modifications.